tests/nosetests/pyanaconda_tests/localization_test.py

```diff
@@ -17,9 +17,191 @@
 #
 
 from pyanaconda import localization
+from pyanaconda.core.constants import DEFAULT_LANG
 from pyanaconda.core.util import execWithCaptureBinary
 import locale as locale_mod
 import unittest
+from unittest.mock import call, patch, MagicMock
+from io import StringIO
+
+
+class LangcodeLocaleParsingTests(unittest.TestCase):
+    def is_valid_test(self):
+        self.assertTrue(localization.is_valid_langcode("en"))
+        self.assertTrue(localization.is_valid_langcode("en_US"))
+        self.assertTrue(localization.is_valid_langcode("cs_CZ.UTF-8@latin"))
+
+    def invalid_raise_decorator_test(self):
+        self.assertRaises(localization.InvalidLocaleSpec, localization.get_native_name, "blah")
+        self.assertRaises(localization.InvalidLocaleSpec, localization.is_supported_locale, "blah")
+
+    def is_supported_test(self):
+        self.assertTrue(localization.is_supported_locale("en"))
+        self.assertTrue(localization.is_supported_locale("en_US"))
+        self.assertTrue(localization.locale_supported_in_console("en"))
+        self.assertTrue(localization.locale_supported_in_console("en_US"))
+
+    def native_name_test(self):
+        self.assertEqual(localization.get_native_name("de"), "Deutsch")
+        self.assertEqual(localization.get_native_name("cs_CZ"), "Čeština (Česko)")
+
+    def available_translations_test(self):
+        self.assertIn("en", localization.get_available_translations())
+
+    def territory_locales_test(self):
+        self.assertIn("en_US.UTF-8", localization.get_territory_locales("US"))
+        self.assertIn("en_GB.UTF-8", localization.get_territory_locales("GB"))
+
+    def locale_keyboards_test(self):
+        self.assertEqual(localization.get_locale_keyboards("en_US"), ["us"])
+        self.assertEqual(localization.get_locale_keyboards("en_GB"), ["gb"])
+
+    def locale_timezones_test(self):
+        self.assertIn("Europe/Oslo", localization.get_locale_timezones("no"))
+
+
+class SetupLocaleTest(unittest.TestCase):
+
+    @patch("pyanaconda.localization.setenv")
+    @patch("pyanaconda.localization.locale_mod.setlocale")
+    @patch("pyanaconda.localization.set_modules_locale")
+    def setup_locale_notext_test(self, set_modules_locale_mock, setlocale_mock, setenv_mock):
+        """Test setup_locale in GUI mode"""
+
+        loc_proxy = MagicMock()
+
+        locale = localization.setup_locale("sk", localization_proxy=loc_proxy)
+
+        loc_proxy.SetLanguage.assert_called_once_with("sk")
+        setenv_mock.assert_called_once_with("LANG", "sk")
+        setlocale_mock.assert_called_once_with(locale_mod.LC_ALL, "sk")
+        set_modules_locale_mock.assert_called_once_with("sk")
+
+        self.assertEqual(locale, "sk")
+
+    @patch("pyanaconda.localization.os.environ", new_callable=dict)
```

Comment:
Shouldn't this be replaced by [patch.dict()](https://docs.python.org/3/library/unittest.mock.html#unittest.mock.patch.dict)?

Reply:
I need to manipulate the dictionary inside the test and patch.dict() gives me no variable to access it.

Reply:
On a further thought, perhaps I am reading the docs wrong, it does return a context manager... let's try.

Reply:
OK, so no variable, but I can access the thing via full path as `localization.os.environ`. Do you feel it is worth the change?

Reply:
Used patch.dict anyway.

Reply:
This was only a suggestion. I'll leave the solution on you.